### PR TITLE
docs: update docs to point to `derivation_of_spk`

### LINKED ISSUE #1042: The Wallet.is_mine() function should return K
**Describe the enhancement**  

The `Wallet.is_mine(&self, script: &[Script])` function currently returns a `bool` representing "whether or not a script is part of this wallet (either internal or external)". It would be more useful to know which `KeychainKind` the script was found for, or `None` if not one of my scripts:

```
pub fn is_mine(&self, script: &[Script]) -> Option<KeychainKind>
```

**Use case**  

A request came up on discord about how to figure out which output is the change. Having `is_mine()` tell you which chain a script is for would make this easy to figure out.

### PR DESCRIPTION
**Pull request by 15IITian**
Updated docs  of `Wallet.is_mine()` function to point `derivation_of_spk` function to find whether a `KeychainKind` exits  for the given `script` or not.

fix #1042

* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

### CODE REVIEW

File: crates/bdk/src/wallet/mod.rs [hunk 1]
```diff
@@ -815,6 +815,8 @@ impl<D> Wallet<D> {
     }
 
     /// Return whether or not a `script` is part of this wallet (either internal or external)
+    /// Use `derivation_of_spk` function to find the `KeychainKind` for the `script`, or
+    /// None if there's no  associated `Keychain`.
```
Comment (evanlinjin) [inline crates/bdk/src/wallet/mod.rs:819]: Please follow rust doc conventions. Ref: https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html#documenting-components

We want a separate section for the more detailed description. Also, for clarity, I would reword it like so: "If you wish to find the `KeychainKind` for the given `script`, use `derivation_of_spk` instead."

In addition to this, please have a link for `derivation_of_spk`.

**Review by evanlinjin — CHANGES_REQUESTED** [review]

File: crates/bdk/src/wallet/mod.rs (hunk 1 above)
Comment (15IITian) [reply crates/bdk/src/wallet/mod.rs:819]: Would it make sense to have  seperate section telling about the  `derivation_of_spk` in detail  and include that point ?

Comment (evanlinjin) [reply crates/bdk/src/wallet/mod.rs:819]: If you want to expand on `derivation_of_spk`, it should be part of the docs on `derivation_of_spk`, not here.

**Comment by evanlinjin** [comment]
Closing because stale.
